go-controller/pkg/node/controllers/egressip/egressip.go

```diff
@@ -0,0 +1,1130 @@
+	expectedIPTableV4Rules := sets.New[string]()
+	expectedIPTableV6Rules := sets.New[string]()
+	egressIPs, err := c.getAllEIPs()
+	if err != nil {
+		return err
+	}
+	node, err := c.nodeLister.Get(c.nodeName)
+	if err != nil {
+		return err
+	}
+	for _, egressIP := range egressIPs {
+		if len(egressIP.Status.Items) == 0 {
+			continue
+		}
+		for _, status := range egressIP.Status.Items {
+			if isValid := isEIPStatusItemValid(status, c.nodeName); !isValid {
+				continue
+			}
+			eIPNet, err := getFullMaskIPNet(status.EgressIP)
+			if err != nil {
+				return err
+			}
+			isOVNManaged, err := util.IsOVNManagedNetwork(node, eIPNet.IP)
+			if err != nil {
+				return fmt.Errorf("failed to determine if EgressIP %s IP %s is OVN managed: %v", egressIP.Name,
+					status.EgressIP, err)
+			}
+			if isOVNManaged {
+				continue
+			}
+			isV6 := eIPNet.IP.To4() == nil
+			found, link, err := findLinkToHostIP(eIPNet.IP, c.v4, c.v6)
+			if err != nil {
+				return fmt.Errorf("failed to find a network to host EgressIP %s IP %s: %v", egressIP.Name,
+					eIPNet.IP.String(), err)
+			}
+			if !found {
+				return fmt.Errorf("expected to find network interface to assign EgressIP %s IP %s", egressIP.Name,
+					status.EgressIP)
+			}
+			linkIdx := link.Attrs().Index
+			linkName := link.Attrs().Name
+			expectedIPRoutes.Insert(getDefaultRoute(linkIdx, isV6).String())
+			expectedAddrs.Insert(getNetlinkAddressWithLabel(eIPNet, linkIdx, linkName).String())
+			namespaceSelector, err := metav1.LabelSelectorAsSelector(&egressIP.Spec.NamespaceSelector)
+			if err != nil {
+				return fmt.Errorf("invalid namespaceSelector for egress IP %s: %v", egressIP.Name, err)
+			}
+			podSelector, err := metav1.LabelSelectorAsSelector(&egressIP.Spec.PodSelector)
+			if err != nil {
+				return fmt.Errorf("invalid podSelector for egress IP %s: %v", egressIP.Name, err)
+			}
+			namespaces, err := c.namespaceLister.List(namespaceSelector)
+			if err != nil {
+				return fmt.Errorf("failed to list namespaces using selector %s to configure egress IP %s: %v",
+					namespaceSelector.String(), egressIP.Name, err)
+			}
+			for _, namespace := range namespaces {
+				namespaceLabels := labels.Set(namespace.Labels)
+				if namespaceSelector.Matches(namespaceLabels) {
+					pods, err := c.podLister.Pods(namespace.Name).List(podSelector)
+					if err != nil {
+						return fmt.Errorf("failed to list pods using selector %s to configure egress IP %s: %v",
+							podSelector.String(), egressIP.Name, err)
+					}
+					for _, pod := range pods {
+						if util.PodCompleted(pod) || util.PodWantsHostNetwork(pod) || len(pod.Status.PodIPs) == 0 {
+							continue
+						}
+						podIPs, err := util.DefaultNetworkPodIPs(pod)
+						if err != nil {
+							return err
+						}
+						for _, ip := range podIPs {
+							// we don't check if its v4 or v6 and if we support it. Controller may have supported a family
+							// previously, but no longer does.
```

Comment:
Shouldn't this consideration be done for the `assigned` ones and not the `expected` ones? There could be v6 stuff for which the controller is not enabled that we should clean up?

Reply:
yes. thank you!